### PR TITLE
[php] Add extern for `version_compare()`

### PR DESCRIPTION
**Pull request by cedx**

### CODE REVIEW

**Comment by back2dos** [comment]
On a slightly more general note: I wonder if it wouldn't make sense to dress up these mixed return types into some more meaningful abstract, that's 1. more self-documenting and 2. easier to use (via a careful selection of all the goodies that can be added to abstracts).

**Comment by cedx** [comment]
I really like this idea, but I'm wondering if it's worth implementing it. In a Haxe project, we should avoid native APIs except when absolutely necessary (for example to optimize performance, or when there is no alternative in Haxe).
The `version_compare()` call could eventually be replaced by the usage of the [`tink_semver` package](https://lib.haxe.org/p/tink_semver/) :wink:
If someone uses a native API, I tend to think he/she is familiar with that API, and therefore already familiar with the mixed return type.

**Comment by RealyUniqueName** [comment]
There is a nice idea proposed by @mockey with `FalseOr<ReturnType>`: https://github.com/HaxeFoundation/haxe/commit/31c5ad3dbfd6c59edf54c0f4c5687f1cee435a66#commitcomment-41468289
I'm going to implement it eventually if he doesn't come with a PR before.

File: std/php/Global.hx [hunk 1]
```diff
@@ -1575,4 +1575,9 @@ extern class Global {
 		@see http://php.net/manual/en/function.http-response-code.php
 	**/
 	static function http_response_code(?response_code:Int):EitherType<Int, Bool>;
+
+	/**
+		@see http://php.net/manual/en/function.version-compare.php
+	**/
+	static function version_compare(version1:String, version2:String, ?comparisonOperator:String):EitherType<Int, Bool>;
```
Comment (RealyUniqueName) [inline std/php/Global.hx]: This should be defined with overloads because the return type depends on a number of arguments.

**Review by RealyUniqueName — CHANGES_REQUESTED** [review]

File: std/php/Global.hx (hunk 1 above)
Comment (CedX) [reply std/php/Global.hx]: Fixed :o)

**Comment by cedx** [comment]
`FalseOr<ReturnType>` would be great: it's such common in PHP APIs.

**Pull request merged**